mobile_verifier/src/boosting_oracles/data_sets.rs

```diff
@@ -0,0 +1,632 @@
+    A: DataSet,
+    B: DataSet,
+    C: DataSet,
+{
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::prelude::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        let handle = tokio::spawn(async move {
+            #[rustfmt::skip]
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => Ok(()),
+                result = self.run() => result,
+            }
+        });
+        Box::pin(
+            handle
+                .map_err(anyhow::Error::from)
+                .and_then(|result| async move { result.map_err(anyhow::Error::from) }),
+        )
+    }
+}
+
+impl
+    DataSetDownloaderDaemon<Footfall<DiskTreeMap>, Landtype<DiskTreeMap>, Urbanization<DiskTreeMap>>
+{
+    pub async fn create_managed_task(
+        pool: PgPool,
+        settings: &Settings,
+        file_upload: FileUpload,
+        new_coverage_object_signal: Receiver<()>,
+    ) -> anyhow::Result<impl ManagedTask> {
+        let (oracle_boosting_reports, oracle_boosting_reports_server) =
+            file_sink::FileSinkBuilder::new(
+                FileType::OracleBoostingReport,
+                settings.store_base_path(),
+                file_upload.clone(),
+                concat!(env!("CARGO_PKG_NAME"), "_oracle_boosting_report"),
+            )
+            .auto_commit(false)
+            .roll_time(Duration::minutes(15))
+            .create()
+            .await?;
+
+        let urbanization: Urbanization<DiskTreeMap> = Urbanization::new();
+        let footfall: Footfall<DiskTreeMap> = Footfall::new();
+        let landtype: Landtype<DiskTreeMap> = Landtype::new();
+        let hex_boost_data = HexBoostData::builder()
+            .footfall(footfall)
+            .landtype(landtype)
+            .urbanization(urbanization)
+            .build()?;
+
+        let data_set_downloader = Self::new(
+            pool,
+            hex_boost_data,
+            FileStore::from_settings(&settings.data_sets).await?,
+            oracle_boosting_reports,
+            settings.data_sets_directory.clone(),
+            new_coverage_object_signal,
+        );
+
+        Ok(TaskManager::builder()
+            .add_task(oracle_boosting_reports_server)
+            .add_task(data_set_downloader)
+            .build())
+    }
+}
+
+impl<A, B, C> DataSetDownloaderDaemon<A, B, C>
+where
+    A: DataSet,
+    B: DataSet,
+    C: DataSet,
+{
+    pub fn new(
+        pool: PgPool,
+        data_sets: HexBoostData<A, B, C>,
+        store: FileStore,
+        oracle_boosting_sink: FileSinkClient,
+        data_set_directory: PathBuf,
+        new_coverage_object_signal: Receiver<()>,
+    ) -> Self {
+        Self {
+            pool,
+            data_sets,
+            store,
+            oracle_boosting_sink,
+            data_set_directory,
+            new_coverage_object_signal,
+        }
+    }
+
+    async fn check_for_new_data_sets(&mut self) -> anyhow::Result<()> {
```

Comment:
I find this iteration of `DataSetDownloaderDaemon` much easier to read through. This is great!